format/tzif/tzif.go

```diff
@@ -0,0 +1,205 @@
+package tzif
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/wader/fq/format"
+	"github.com/wader/fq/pkg/decode"
+	"github.com/wader/fq/pkg/interp"
+	"github.com/wader/fq/pkg/scalar"
+)
+
+func init() {
+	interp.RegisterFormat(decode.Format{
+		Name:        format.TZIF,
+		Description: "Time Zone Information Format",
+		DecodeFn:    decodeTZIF,
+		Groups:      []string{format.PROBE},
+	})
+}
+
+func decodeTZIF(d *decode.D, _ any) any {
+	d.Endian = decode.BigEndian
+
+	v1h := decodeTZifHeader(d, "v1header")
+	decodeTZifDataBlock(d, v1h, 1, "v1datablock")
+
+	if v1h.ver >= 2 {
+		v2h := decodeTZifHeader(d, "v2plusheader")
+		decodeTZifDataBlock(d, v2h, 2, "v2plusdatablock")
+		decodeTZifFooter(d)
+	}
+
+	return nil
+}
+
+type tzifHeader struct {
+	magic    uint32
+	ver      uint8
+	isutcnt  uint32
+	isstdcnt uint32
+	leapcnt  uint32
+	timecnt  uint32
+	typecnt  uint32
+	charcnt  uint32
+}
+
+var versionToSymMapper = scalar.UToSymStr{
+	0x00: "1",
+	0x32: "2",
+	0x33: "3",
+}
+
+func decodeTZifHeader(d *decode.D, name string) tzifHeader {
+	var h tzifHeader
+
+	d.FieldStruct(name, func(d *decode.D) {
+		h.magic = uint32(d.FieldU32("magic", scalar.ActualHex, d.AssertU(0x545a6966)))
+		h.ver = uint8(d.FieldU8("ver", d.AssertU(0x00, 0x32, 0x33), scalar.ActualHex, versionToSymMapper))
+		d.FieldRawLen("reserved", 15*8)
+		h.isutcnt = uint32(d.FieldU32("isutcnt"))
+		h.isstdcnt = uint32(d.FieldU32("isstdcnt"))
+		h.leapcnt = uint32(d.FieldU32("leapcnt"))
+		h.timecnt = uint32(d.FieldU32("timecnt"))
+		h.typecnt = uint32(d.FieldU32("typecnt"))
+		h.charcnt = uint32(d.FieldU32("charcnt"))
+	})
+
+	if h.isutcnt != 0 && h.isutcnt != h.typecnt {
+		d.Fatalf("invalid isutcnt")
+	}
+	if h.isstdcnt != 0 && h.isstdcnt != h.typecnt {
+		d.Fatalf("invalid isstdcnt")
+	}
+	if h.typecnt == 0 {
+		d.Fatalf("invalid typecnt")
+	}
+	if h.charcnt == 0 {
+		d.Fatalf("invalid charcnt")
+	}
+
+	return h
+}
+
+type unixTimeToStrMapper struct {
+}
+
+func (m unixTimeToStrMapper) MapScalar(s scalar.S) (scalar.S, error) {
+	i, ok := s.Value().(int64)
+	if !ok {
+		return s, fmt.Errorf("expected int64 but got %T", s.Value())
+	}
+
+	s.Sym = time.Unix(i, 0).UTC().Format(time.RFC3339)
+	return s, nil
+}
```

Comment:
There is a `scalar.Fn` if you want to do adhoc mappers using a function

Reply:
Thank you for letting me know about it. Updated @ [2fd4826](https://github.com/wader/fq/pull/498/commits/2fd48261e55f39829839c50551bbfaac58633c1d)